pkg/fileservice/mem_cache.go

```diff
@@ -160,15 +176,30 @@ func (m *MemCache) Update(
 			Offset: entry.Offset,
 			Size:   entry.Size,
 		}
+
+		var isNewEntry bool
 		if async {
 			obj := entry.ObjectBytes // copy from loop variable
 			objSize := entry.ObjectSize
 			m.ch <- func() {
+				//TODO: Need advise on how to handle async scenario?
```

Comment:
Need suggestion on handling async scenario. Are we using it right now?

Reply:
I think it's the same to sync.

Reply:
Got it.

I updated it like this

```go
key := IOVectorCacheKey{
	Path:   path.File,
	Offset: entry.Offset,
	Size:   entry.Size,
}

var isNewEntry bool
if async {
	obj := entry.ObjectBytes // copy from loop variable
	objSize := entry.ObjectSize
	m.ch <- func() {
		m.objCache.Set(key, obj, objSize, vector.Preloading)

		// Update overlap checker when new key-interval is inserted into the cache.
		// If we are replacing the data for an existing key, we don't have issue of wasted memory space.
		if isNewEntry {
			if err = m.overlapChecker.Insert(key.Path, key.Offset, key.Offset+key.Size); err != nil {
				panic(err)
			}
		}
	}
} else {
	isNewEntry = m.objCache.Set(key, entry.ObjectBytes, entry.ObjectSize, vector.Preloading)

	// Update overlap checker when new key-interval is inserted into the cache.
	// If we are replacing the data for an existing key, we don't have issue of wasted memory space.
	if isNewEntry {
		if err = m.overlapChecker.Insert(key.Path, key.Offset, key.Offset+key.Size); err != nil {
			panic(err)
		}
	}
}

```

Reply:
I will need to make this concurrent.

Reply:
Done.